src/NHibernate/Tool/hbm2ddl/SchemaExport.cs

```diff
@@ -203,10 +203,7 @@ private void ExecuteSql(DbCommand cmd, string sql)
 		{
 			if (dialect.SupportsSqlBatches)
 			{
-				var objFactory = Environment.ObjectsFactory;
-				ScriptSplitter splitter = (ScriptSplitter)objFactory.CreateInstance(typeof(ScriptSplitter), sql);
-
-				foreach (string stmt in splitter)
+				foreach (var stmt in new ScriptSplitter(sql))
```

Comment:
As discussed on #1758, the `ScriptSplitter` does not expose anything overridable, and so it cannot benefit from being instantiated by the object factory: it cannot supply a custom implementation.